megatron/core/datasets/utils.py

```diff
@@ -74,7 +74,7 @@ def get_blend_from_list(
             try:
                 weight = float(rwpd)
             except ValueError:
-                weight = None
+                weight = rwpd
```

Comment:
So in case the weight is not a number, we assume its a path?
Won't it affect the behaviour elsewhere?

Reply:
(if the code expects Nones, like the line below)

Reply:
Yes, this was a bit harcoded. I changed this and think it's fine now